api/v1alpha1/variantautoscaling_types.go

```diff
@@ -11,13 +11,41 @@ type VariantAutoscalingSpec struct {
 	// +kubebuilder:validation:Required
 	ModelID string `json:"modelID"`
 
+	// VariantID uniquely identifies this variant (model + accelerator + acceleratorCount combination).
+	// This is a business identifier that may contain slashes, dots, and mixed case.
+	// Format: {modelID}-{accelerator}-{acceleratorCount}
+	// Example: "meta/llama-3.1-8b-A100-4"
+	//
+	// Note: VariantID (variant_id) is distinct from the VariantAutoscaling resource name (variant_name):
+	//   - variant_id (this field): Business identifier, may contain non-K8s-compliant characters
+	//   - variant_name (resource.Name): Kubernetes resource name (DNS-1123 compliant)
+	//
+	// Both identifiers are exposed as Prometheus labels for flexible querying:
+	//   - Use variant_name to query by Kubernetes resource (typically matches Deployment name)
+	//   - Use variant_id to query by business identifier (model/variant naming)
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^.+-[A-Za-z0-9]+-[1-9][0-9]*$`
```

Comment:
Isn't this pattern restrictive? Should we make it more flexible? for example for llama-8b-h100-1